src/action.cpp

```diff
@@ -556,12 +556,12 @@ action_id handle_action_menu()
     const input_context ctxt = get_default_mode_input_context();
     std::string catgname;
 
-#define REGISTER_ACTION(name) entries.push_back(uimenu_entry(name, true, hotkey_for_action(name), \
+#define REGISTER_ACTION(name) entries.emplace_back(uimenu_entry(name, true, hotkey_for_action(name), \
```

Comment:
We have warnings about not using emplace? That's bizarre, since this case should be handled by copy elision.

Reply:
Most of these are actually what my IDE flags as a "weak warning" (i.e., not something that compilers are going to usually warn you about, but rather cases where there's something better out there, bad practices, new and useful features, etc.). In this case there's almost zero cases where emplace_back won't be better than push_back in terms of reliability across platforms and performance, so it gets flagged as such.

Reply:
My understanding is emplace can prevent copy elision, which ends up being a superior solution.
Also depending on the container, emplace will accept a wide range of arguments, where push will only accept one, so it's more precise.

Reply:
Everything I'm finding online in regards to this is telling me that even with copy elision ```emplace_back``` will perform at worst equivalent to ```push_back``` (because emplace is already removing the copy that copy elision would be factoring out). I've found at least a few places where people have come right out and said that performance of emplace will always be equal or less than that of push_back in C++11 at least, so it's good as a default unless you have reasons not to use it, because worst case scenario it literally devolves to the same behind the scene calls as push_back (and thus triggers the same copy elision) and better case it constructs in place which obviates the need for copy elision in the first place.